hub/core/chunk_engine.py

```diff
@@ -644,10 +645,57 @@ def _create_new_chunk(self, register=True) -> BaseChunk:
         chunk._update_tensor_meta_length = register
         if self.active_appended_chunk is not None:
             self.write_chunk_to_storage(self.active_appended_chunk)
-
         self.active_appended_chunk = chunk
         return chunk
 
+    def clear(self):
+        """Clears all samples and cachables."""
+        self.cache.check_readonly()
+
+        commit_id = self.version_state["commit_id"]
+
+        enc = self.chunk_id_encoder
+        n_chunks = self.num_chunks
+        chunk_names = [enc.get_name_for_chunk(i) for i in range(n_chunks)]
+        chunk_keys = [get_chunk_key(self.key, name, commit_id) for name in chunk_names]
+
+        enc_key = get_chunk_id_encoder_key(self.key, commit_id)
+        try:
+            self._chunk_id_encoder = None
+            del self.meta_cache[enc_key]
+        except KeyError:
+            pass
+
+        for key in chunk_keys:
+            try:
+                del self.cache[key]
```

Comment:
why don't we leverage the clear method of providers for this? will help us clean out entire folders much faster.